src/rgw/rgw_rest_log.cc

```diff
@@ -323,20 +323,18 @@ void RGWOp_MDLog_Unlock::execute() {
 }
 
 void RGWOp_MDLog_Notify::execute() {
-  char *data;
-  int len = 0;
 #define LARGE_ENOUGH_BUF (128 * 1024)
-  int r = rgw_rest_read_all_input(s, &data, &len, LARGE_ENOUGH_BUF);
+  auto [r, data] = rgw_rest_read_all_input(s, LARGE_ENOUGH_BUF);
   if (r < 0) {
     http_ret = r;
     return;
   }
 
-  ldout(s->cct, 20) << __func__ << "(): read data: " << string(data, len) << dendl;
+  char* buf = data.c_str();
+  ldout(s->cct, 20) << __func__ << "(): read data: " << buf << dendl;
 
   JSONParser p;
-  r = p.parse(data, len);
-  free(data);
+  r = p.parse(buf, data.length());
```

Comment:
does JSONParser not have a buffer::list version?

Reply:
Will check

Reply:
It doesn't

Reply:
I can add a buffer::list input stream for it. I'll do that and have a PR for it.

(No need to hold this PR up for it, though, I don't think.)